### PR TITLE
Record more complete information about contestEntry

### LINKED ISSUE #213: More complete ShowcaseRankings
Today after I put my shiny Kyogre in a showcase, I was furious to find out that Golbat (and therefore RM) does not support parsing whether an entered Pokemon was shiny. Therefore, I propose the following additions:

1. Shiny (boolean)
2. Temp evolution (mega can be in fact entered in this showcase)
3. Alignment (shadow/purified/normal)
4. Badge (best buddy/none)

Thoughts?

### PR DESCRIPTION
**Pull request by Mygod**
Fixes #213. Need help testing.

### CODE REVIEW

**Comment by Fabio1988** [comment]
You really want to flex? :wink: 

Why is your approach to store the display completely different from #217 ?

**Comment by Mygod** [comment]
It's the same? 👀 Some fields are not there for backwards compatibility reasons, like `pokemon_id`.

**Comment by Mygod** [comment]
> You really want to flex? 😉

What's the point of grinding if I can't flex? 😉

**Comment by Mygod** [comment]
Can we get some poggers on this PR?

**Review by Fabio1988 — APPROVED** [review]

**Comment by Fabio1988** [comment]
looks solid, you tested this?

**Comment by Mygod** [comment]
Actually I need to finish RM support for gym defender first. Please standby. :)

**Comment by Mygod** [comment]
Can be merged now!

File: decoder/pokestop.go [hunk 1]
```diff
@@ -638,12 +644,23 @@ func (stop *Pokestop) updatePokestopFromGetPokemonSizeContestEntryOutProto(conte
...
-			Gender:    int(entry.GetPokemonDisplay().Gender),
+			Rank:                  int(rank),
+			Score:                 entry.GetScore(),
+			PokemonId:             int(entry.GetPokedexId()),
+			Form:                  int(entry.GetPokemonDisplay().Form),
+			Costume:               int(entry.GetPokemonDisplay().Costume),
+			Gender:                int(entry.GetPokemonDisplay().Gender),
+			Shiny:                 entry.GetPokemonDisplay().Shiny,
+			TempEvolution:         int(entry.GetPokemonDisplay().CurrentTempEvolution),
+			TempEvolutionFinishMs: entry.GetPokemonDisplay().TemporaryEvolutionFinishMs,
```
Comment (Fabio1988) [inline decoder/pokestop.go:655]: this might be ns, not ms.... had the same issue on processing power stations... nia is not consistent with it.

Comment (Mygod) [reply decoder/pokestop.go:655]: Maybe we can check the proto dump to confirm? :eyes:

Comment (Mygod) [reply decoder/pokestop.go:655]: Regardless I'm in favor of keeping proto as is without doing any unit conversions. Those extra data might be useful for reverse engineering or something. We are not saving any storage/bandwidth anyways.

**Comment by Fabio1988** [comment]
```
{"total_entries":9,"last_update":1726506754,"contest_entries":[{"rank":1,"score":1127.381020222036,"pokemon_id":428,"form":0,"costume":0,"gender":1,"shiny":true,"temp_evolution":1,"temp_evolution_finish_ms":1721221702374,"alignment":0,"badge":0,"location_card":0},{"rank":2,"score":1110.9649792598755,"pokemon_id":531,"form":0,"costume":0,"gender":1,"shiny":false,"temp_evolution":0,"temp_evolution_finish_ms":0,"alignment":0,"badge":0,"location_card":0},{"rank":3,"score":1106.0886862076218,"pokemon_id":352,"form":0,"costume":0,"gender":2,"shiny":false,"temp_evolution":0,"temp_evolution_finish_ms":0,"alignment":0,"badge":0,"location_card":0}]}
```

@Mygod seems to be ns, not ms, right? :)

**Comment by Mygod** [comment]
image

Seems like ms to me?

**Comment by Fabio1988** [comment]
> ```
> {"total_entries":9,"last_update":1726506754,"contest_entries":[{"rank":1,"score":1127.381020222036,"pokemon_id":428,"form":0,"costume":0,"gender":1,"shiny":true,"temp_evolution":1,"temp_evolution_finish_ms":1721221702374,"alignment":0,"badge":0,"location_card":0},{"rank":2,"score":1110.9649792598755,"pokemon_id":531,"form":0,"costume":0,"gender":1,"shiny":false,"temp_evolution":0,"temp_evolution_finish_ms":0,"alignment":0,"badge":0,"location_card":0},{"rank":3,"score":1106.0886862076218,"pokemon_id":352,"form":0,"costume":0,"gender":2,"shiny":false,"temp_evolution":0,"temp_evolution_finish_ms":0,"alignment":0,"badge":0,"location_card":0}]}
> ```
> 
> @Mygod seems to be ns, not ms, right? :)

damn you are right... too late here :D

**Comment by Mygod** [comment]
By the way I just checked and it seems like dynamax pokemon can enter showcases but displays as normal Pokemon in the game. Should we add this information now or hold out for later when it matters?

**Comment by Fabio1988** [comment]
hold out for later... right now even dynamax pokemon are not displayed as dynamax in gyms :D

**Comment by Mygod** [comment]
> Regardless I'm in favor of keeping proto as is without doing any unit conversions. Those extra data might be useful for reverse engineering or something. We are not saving any storage/bandwidth anyways.

Umm... We already had this conversation 2 weeks ago...?

**Comment by Fabio1988** [comment]
> > Regardless I'm in favor of keeping proto as is without doing any unit conversions. Those extra data might be useful for reverse engineering or something. We are not saving any storage/bandwidth anyways.
> 
> Umm... We already had this conversation 2 weeks ago...?

It's not about storage, just about consistency.. e.g. when RM wants to display it...  :)
But agree it's the same in gym display, so fine

**Comment by Mygod** [comment]
Thank you. :)

**Pull request merged**